logprep/processor/generic_adder/rule.py

```diff
@@ -143,9 +143,23 @@ class Config(FieldManagerRule.Config):
         first existing file by setting :code:`generic_adder.only_first_existing_file: true`.
         In that case, only one file must exist."""
 
+        _base_add: dict = field(default={}, eq=False)
+        """Stores original add fields for future refreshes of getters"""
+
         # pylint: enable=anomalous-backslash-in-string
 
+        def _refresh_add(self):
+            self.add = copy.deepcopy(self._base_add)
+            self._add_from_path()
+
         def __attrs_post_init__(self):
+            self._base_add = copy.deepcopy(self.add)
+
+            for add_file in self.add_from_file:  # pylint: disable=not-an-iterable
+                getter = GetterFactory.from_string(add_file)
+                if isinstance(getter, RefreshableGetter):
+                    getter.add_callback(self._refresh_add)
+
             if self.add_from_file:
                 self._add_from_path()
```

Comment:
```suggestion
            if self.add_from_file:
                for add_file in self.add_from_file:  # pylint: disable=not-an-iterable
                    getter = GetterFactory.from_string(add_file)
                    if isinstance(getter, RefreshableGetter):
                        getter.add_callback(self._refresh_add)
                self._add_from_path()
```

Seems logical to me to group the file-related stuff when doing the conditional stuff either way.